backend/src/ee/routes/v1/ldap-router.ts

```diff
@@ -44,26 +46,94 @@ export const registerLdapRouter = async (server: FastifyZodProvider) => {
     });
   };
 
+  interface LDAPConfig {
+    id: string;
+    organization: string;
+    isActive: boolean;
+    url: string;
+    bindDN: string;
+    bindPass: string;
+    searchBase: string;
+    groupSearchBase: string;
+    groupSearchFilter: string;
+    caCert: string;
+  }
+
   passport.use(
     new LdapStrategy(
       getLdapPassportOpts as any,
       // eslint-disable-next-line
       async (req: IncomingMessage, user, cb) => {
         try {
-          const { isUserCompleted, providerAuthToken } = await server.services.ldap.ldapLogin({
-            externalId: user.uidNumber,
-            username: user.uid,
-            firstName: user.givenName,
-            lastName: user.sn,
-            emails: user.mail ? [user.mail] : [],
-            relayState: ((req as unknown as FastifyRequest).body as { RelayState?: string }).RelayState,
-            orgId: (req as unknown as FastifyRequest).ldapConfig.organization
+          const ldapConfig = (req as unknown as FastifyRequest).ldapConfig as LDAPConfig;
```

Comment:
Just a suggestion try to keep the async await everywhere and avoid mixing up async await with then catch.

It's a bit confusing to read as you have to switch between two mental models